### ISSUE TITLE
Index column selection with camelCasing

### ISSUE THREAD

**Issue by eltonio450** [Kononnable/typeorm-model-generator#103]
Hello !

Thanks for this tool, it helps us so much in our migration :)

I generated my types with the --cp (and --ce, --cf) options as camel (bc we target GraphQL). I have indices on my table. (EDIT: postgres driver)

The actual output is:

```
import {Column, Entity, Index, JoinColumn, JoinTable, ManyToMany, ManyToOne, OneToMany, OneToOne, PrimaryColumn, PrimaryGeneratedColumn, RelationId} from "typeorm";

@Entity("token_sessions", {schema: "public"})
@Index("index_token_sessions_on_auth_token", ["auth_token"], {unique: true})
@Index("index_token_sessions_on_user_id", ["user_id"])
export class tokenSessions {

    @PrimaryGeneratedColumn({
        name: "id",
        })
    public id: string;


    @Column("integer", {
        nullable: true,
        name: "user_id",
        })
    public userId: number | null;

    @Column("character varying", {
        nullable: true,
        default: "''::character varying",
        name: "auth_token",
        })
    public authToken: string | null;
}
```

Expected output:

```
import {Column, Entity, Index, JoinColumn, JoinTable, ManyToMany, ManyToOne, OneToMany, OneToOne, PrimaryColumn, PrimaryGeneratedColumn, RelationId} from "typeorm";

@Entity("token_sessions", {schema: "public"})
@Index("index_token_sessions_on_auth_token", ["authToken"], {unique: true})
@Index("index_token_sessions_on_user_id", ["userId"])
export class tokenSessions {

    @PrimaryGeneratedColumn({
        name: "id",
        })
    public id: string;


    @Column("integer", {
        nullable: true,
        name: "user_id",
        })
    public userId: number | null;

    @Column("character varying", {
        nullable: true,
        default: "''::character varying",
        name: "auth_token",
        })
    public authToken: string | null;
}
```

Line 4 and 5, the second argument should be camel cased I think. I am not 100% sure bc I am not familiar with Typeorm, but manually correcting the colmun fix the issue at runtime

Should I open a PR ?

Second question/problem (if it's a problem, I'll open an issue): my PrimaryGeneratedColumn is an Int in my database. Why is it generated as a string ?

Thank you!

Antoine

**Comment by eltonio450** [comment]
Ok for the second question, it turns out our ids are big int, which are transformed as string, so it's not not a bug https://github.com/tgriesser/knex/issues/387

**Comment by eltonio450** [comment]
@Kononnable here is the fix for the issue:

````diff
@Entity("{{EntityName}}"{{#Schema}},{schema:"{{.}}"}{{/Schema}})
{{#Indexes}}{{^isPrimaryKey}}@Index("{{name}}",[{{#columns}}"{{name}}",{{/columns}}]{{#isUnique}},{unique:true}{{/isUnique}})
{{#Indexes}}{{^isPrimaryKey}}@Index("{{name}}",[{{#columns}}"{{toPropertyName name}}",{{/columns}}]{{#isUnique}},{unique:true}{{/isUnique}})
{{/isPrimaryKey}}{{/Indexes}}export class {{toEntityName EntityName}} {
{{#Columns}}
```

tried to write a test but not did not succeed in starting it :(. Lmk if I can help :)

**Comment by Kononnable** [comment]
You're right. Column names in indices should also get proper casing.
Writing test for naming changes is hard right now. I plan to add another step in model generation to isolate such changes(and make test for them) but it is put on hold - main typeorm repo looks like it need more love now :)

**Comment by Kononnable** [comment]
Fixed in `typeorm-model-generator@0.2.22`.
